modules/hdf5_handler/HDF5CFByte.cc

```diff
@@ -49,7 +48,8 @@ HDF5CFByte::HDF5CFByte(const string &n, const string &d,const string &d_f) : Byt
 
 BaseType *HDF5CFByte::ptr_duplicate()
 {
-    return new HDF5CFByte(*this);
+    auto HDF5CFByte_unique = make_unique<HDF5CFByte>(*this);
```

Comment:
For cases like this, it's probably better to not make a unique pointer since there's nothing between the constructor and the release.

Reply:
I will keep this because
1)  Since sonar cloud complains this as code smells. If using the unique pointer, a code smell is reduced. 
2) I've done this before and you approved it. It will be time consuming for me to go back to change them since there are a lot of them.